BuildScript.Nuke/GenerateSbom/SolutionSbomGenerator.cs

```diff
@@ -0,0 +1,177 @@
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Tools.Npm;
+using Nuke.Common.Tools.PowerShell;
+using SbomCleaner.Library;
+using Serilog;
+
+namespace Remotion.BuildScript.GenerateSbom;
+
+public class SolutionSbomGenerator: ISbomGenerator
+{
+  public SbomGeneration SbomGeneration => SbomGeneration.ForSolution;
+
+  private readonly FilterList _packageBlackList;
+  private readonly FilterList _projectBlacklist;
+
+  private readonly string _githubUsername;
+  private readonly string _githubAccessToken;
+
+  private readonly AbsolutePath _outputFile;
+
+  private readonly Solution _solution;
+  private readonly AbsolutePath _workingDirectory;
+
+  private readonly AbsolutePath? _pathToPackageJson;
+
+  public SolutionSbomGenerator (
+      Solution solution,
+      AbsolutePath workingDirectory,
+      AbsolutePath outputFile,
+      FilterList packageBlackList,
+      FilterList projectBlacklist,
+      AbsolutePath? pathToPackageJson,
+      string githubUsername,
+      string githubAccessToken)
+  {
+    _solution = solution;
+    _workingDirectory = workingDirectory;
+    _outputFile = outputFile;
+    _projectBlacklist = projectBlacklist;
+    _packageBlackList = packageBlackList;
+    _pathToPackageJson = pathToPackageJson;
+    _githubUsername = githubUsername;
+    _githubAccessToken = githubAccessToken;
+  }
+
+  public async Task Generate (Tool cycloneDxTool)
+  {
+    var uncleanedSbomFilename = "uncleaned-sbom.xml";
+
+    Directory.CreateDirectory(_workingDirectory);
+
+    cycloneDxTool.Invoke(
+        arguments:
+        $"-o . --disable-package-restore --exclude-dev --exclude-test-projects -gu {_githubUsername} -gt {_githubAccessToken} -f \"{uncleanedSbomFilename}\" {_solution.Path}",
+        workingDirectory: _workingDirectory
+    );
+
+    Log.Information("Cleaning Sbom...");
+
+    var cleanedSbomFileName = "cleaned-sbom.xml";
+    var cleaner = new Cleaner();
+
+    cleaner.CleanSolutionSbom(
+        _workingDirectory / uncleanedSbomFilename,
+        _solution.Path,
+        _packageBlackList,
+        _projectBlacklist,
+        _workingDirectory / cleanedSbomFileName);
+
+    Log.Information("Created cleaned Sbom '{_outputFile}'.", _workingDirectory / cleanedSbomFileName);
+
+    if (_pathToPackageJson == null)
+    {
+      Log.Information("No package json file specified, will therefore not create combined Sbom.");
+      return;
+    }
```

Comment:
I'm not sure we're setting `_outputFile` correctly when we are exiting here. I think it is only populated with the correct data when we do a dotnet and npm comine.